### ISSUE TITLE
Feature request: HierarchicalMenu items with fixed "breadcrumb" value

### ISSUE THREAD

**Issue by zzal** [algolia/react-instantsearch#2389]
Hi there!

I'm using the `transformItems` modifier on a 3-levels deep HierarchicalMenu to ensure the editor's custom sorting of the items is respected.
I'm using a reference array in my transformItems function that looks like this:
```
[
  "Foo",
  "Foo > Bar",
  "Foo > Aoc",
  "Quuz",
  "Quuz > Ham",
  "Quuz > Ham > Xor",
  "Quuz > Ham > Egg",
  "Quuz > Bar",
]
```

I wish I could use each item's `value` property but it changes overtime (when it or one of its children is selected).
I cannot use the item's `label` since it's just a partial information. (See the "Bar" items in my example, which appears multiple times with different sorting positions)

It would be very welcomed if each HierarchicalMenu item had a reliable, immutable value that I could use for sorting reference. Like: 
```
{
  label: "Bar",
  count: 10,
  isRefined: true,
  value: "Foo",
  breadcrumb: "Foo > Bar"
}
```

Or is there a way to hack/extend the `connectHierarchicalMenu` connector to add it my self?
Thanks!

**Comment by tkrugg** [comment]
Hi @zzal. Thanks for reaching out.

I'm not sure I understand what you mean here:

> I wish I could use each item's value property but it changes overtime (when it or one of its children is selected).

Can you elaborate on what situations the value of an item can change?

**Comment by zzal** [comment]
Yes, of course:

The value of each item in a HierarchicalMenu is used to set the `currentRefinement` of the component. So when you click an item (say for example, `Foo > Bar` in my original post), its value changes so the next time this item is clicked it will be unselected, so the value of the item is set to its parent element (`Foo`).

Look at the example below (should be an animated GIF): 
I have added **the corresponding `value` of each item, printed in pink** on the right of its label so you can see what I mean (yeah, sometimes the value is too long, it's wrapping to the next line)

![HierarchicalMenu-on-select-value-change](https://user-images.githubusercontent.com/110392/57097774-9a637d00-6ce6-11e9-8077-7da19684e145.gif)

Does that make sense with those clarifications?

**Comment by tkrugg** [comment]
Ok I think I understand it. Let me rephrase it to make sure we're on the same page.
You want to have an immutable `.breadcrumb` property on each item to allow sorting in transformItems.

Adding the breadcrumb property should be possible in transformItems too.

```jsx
function addBreadcrumb(items, parent = null) {
  for (let item of items) {
    item.breadcrumb = (parent ? parent : "") + " > " + item.label;
    if (item.items) {
      addBreadcrumb(item.items, item.breadcrumb);
    }
  }
}

function transformItems(items) {
  addBreadcrumb(items);

  console.log(items);
  return items;
}

<HierarchicalMenu
  attributes={[
    "hierarchicalCategories.lvl0",
    "hierarchicalCategories.lvl1",
    "hierarchicalCategories.lvl2"
  ]}
  transformItems={transformItems}
/>
```

Live example is here: https://codesandbox.io/s/reactinstantsearchapp-hy5by.
Does this help?

**Comment by samouss** [comment]
I'm closing this issue because it's inactive since a while. Feel free to open a new one!
